modules/runner/src/integrationTest/java/org/apache/ignite/internal/table/ItTransactionConflictTest.java

```diff
@@ -145,6 +164,581 @@ public void testMultipleRecoveryRequestsIssued() throws Exception {
+                .findFirst().get();
+
+        log.info("Transaction commit partition is determined [node={}].", commitPartNode.name());
+
+        IgniteImpl txCrdNode = IntStream.range(1, initialNodes()).mapToObj(this::node).filter(n -> !leaseholder.equals(n.name()))
+                .findFirst().get();
+
+        log.info("Transaction coordinator is chosen [node={}].", txCrdNode.name());
+
+        InternalTransaction orphanTx = (InternalTransaction) createRwTransaction(txCrdNode);
+
+        CompletableFuture<TxFinishReplicaRequest> finishRequestCaptureFut = new CompletableFuture<>();
+        AtomicReference<String> targetName = new AtomicReference<>();
+
+        // Intercept the commit message, prevent it form being sent. We will kill this node anyway.
+        txCrdNode.dropMessages((nodeName, msg) -> {
+            if (msg instanceof TxFinishReplicaRequest) {
+                var finishTxMsg = (TxFinishReplicaRequest) msg;
+
+                finishRequestCaptureFut.complete(finishTxMsg);
+                targetName.set(nodeName);
+
+                return true;
+            }
+
+            return false;
+        });
+
+        // Initiate commit.
+        orphanTx.commitAsync();
+
+        assertThat(finishRequestCaptureFut, willCompleteSuccessfully());
+
+        // Stop old coordinator.
+        String txCrdNodeId = txCrdNode.id();
+
+        txCrdNode.stop();
+
+        assertTrue(waitForCondition(
+                () -> node(0).clusterNodes().stream().filter(n -> txCrdNodeId.equals(n.id())).count() == 0,
+                10_000)
+        );
+
+        // The state on the commit partition is still PENDING.
+        assertEquals(TxState.PENDING, txVolatileState(commitPartNode, orphanTx.id()));
+
+        IgniteImpl newTxCoord = node(0);
+
+        runRwTransactionNoError(newTxCoord, newTxCoord.transactions().begin());
+
+        assertTrue(waitForCondition(() -> txStoredState(commitPartNode, orphanTx.id()) == TxState.ABORTED, 10_000));
+
+        assertTrue(waitForCondition(() -> txStoredMeta(commitPartNode, orphanTx.id()).locksReleased(), 10_000));
+
+        CompletableFuture<NetworkMessage> commitRequest =
+                messaging(commitPartNode).invoke(targetName.get(), finishRequestCaptureFut.join(), 3000);
+
+        assertThat(commitRequest, willCompleteSuccessfully());
+
+        NetworkMessage response = commitRequest.join();
+
+        assertInstanceOf(ErrorTimestampAwareReplicaResponse.class, response);
+
+        ErrorTimestampAwareReplicaResponse errorResponse = (ErrorTimestampAwareReplicaResponse) response;
+
+        assertInstanceOf(TransactionAlreadyFinishedException.class, ExceptionUtils.unwrapCause(errorResponse.throwable()));
+
+        assertTrue(waitForCondition(() -> txStoredState(commitPartNode, orphanTx.id()) == TxState.ABORTED, 10_000));
+    }
+
+    @Test
+    public void testRecoveryIsTriggeredOnce() throws Exception {
```

Comment:
this test is flaky

```
org.opentest4j.AssertionFailedError: 
Expected :true
Actual   :false
<Click to see difference>


	at org.junit.jupiter.api.AssertionFailureBuilder.build(AssertionFailureBuilder.java:151)
	at org.junit.jupiter.api.AssertionFailureBuilder.buildAndThrow(AssertionFailureBuilder.java:132)
	at org.junit.jupiter.api.AssertTrue.failNotTrue(AssertTrue.java:63)
	at org.junit.jupiter.api.AssertTrue.assertTrue(AssertTrue.java:36)
	at org.junit.jupiter.api.AssertTrue.assertTrue(AssertTrue.java:31)
	at org.junit.jupiter.api.Assertions.assertTrue(Assertions.java:180)
	at org.apache.ignite.internal.table.ItTransactionConflictTest.testRecoveryIsTriggeredOnce(ItTransactionConflictTest.java:652)
```
on the line
`
assertTrue(waitForCondition(() -> txStoredState(commitPartNode, rwTx2Id) == TxState.COMMITTED, 10_000));
`